pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -227,6 +231,21 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 		return nil
 	}
 
+	// Ensure that the parameters from the PipelineRun are overriding Pipeline parameters with the same type.
```

Comment:
Just a though: I'm not quite sure how the separation of concerns might be between the validation webhook and the reconciler is meant to be exactly, but it feels to me that this is something that we could verify when the PipelineRun is sent to the API instead of accepting it a failing it here on the parameter types?
I don't mind if this stays like this, but if it makes sense at all we might consider moving this to the webhook.

Reply:
Oh interesting! In my head, I see webhook validation as validation that only uses information on that isolated YAML file being submitted (e.g invalid types, invalid variable interpolation strings), while reconciler validation brings in outside resources and evaluates the collective context. 

In this case, since the validation requires finding the referenced pipeline resource, I put it in the reconciler -- this particular test verifies that, for example, a pipelinerun parameter *string* value is not overriding a corresponding pipeline's parameter *array* value. This seems really similar to the validation done with `GetResourcesFromBindings` (`will validate that all PipelineResources declared in Pipeline p are bound in PipelineRun pr`), which is also in the reconciler.

Writing this up, I think my decision makes sense, but let me know if it doesn't!

Reply:
@afrittoli this is meant to be a reply to your comment above. GitHub is being weird though and showing it as a separate comment so I'm tagging you.